### PR TITLE
Add user quest qr code

### PR DESCRIPTION
**Pull request by ttoino**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -155,4 +159,29 @@ public function edit(Request $request)
...
+    {
+        Gate::authorize('participant');
+
+        /** @var Participant */
+        $user = $request->user()->usertype;
+
+        for ($i = 0; $i < 100; $i++) {
+            $code = Str::random(10);
+
+            if (Participant::where('quest_code', $code)->exists()) {
```
Comment (Naapperas) [inline app/Http/Controllers/UserController.php:178]: Please add a comment explaining this: it should mention that the qr code is unique on the participant

File: resources/js/Components/Profile/InfoCard.vue [hunk 2]
```diff
@@ -81,33 +83,43 @@ const iconColor: Record<string, string> = {
...
+                </svg>
+            </button>
+            <QRCode
+                v-if="
+                    user?.usertype_type == 'App\\Models\\Participant' &&
+                    user.usertype
+                "
+                :participant="user.usertype"
+            ></QRCode>
+            <ModalsContainer></ModalsContainer>
```
Comment (Naapperas) [inline resources/js/Components/Profile/InfoCard.vue]: Isn't this already rendered by the AppLayout?

**Review by Naapperas — CHANGES_REQUESTED** [review]

**Review by pedronunes19 — DISMISSED** [review]
squash commit

**Review by CiscoPr — APPROVED** [review]

**Review by pedrosilva17 — APPROVED** [review]

**Review by Naapperas — APPROVED** [review]
LGTM

**Pull request merged**